cognite/client/_api_client.py

```diff
@@ -77,41 +77,55 @@
 
 class APIClient:
     _RESOURCE_PATH: str
-    # TODO: When Cognite Experimental SDK is deprecated, remove frozenset in favour of re.compile:
-    _RETRYABLE_POST_ENDPOINT_REGEX_PATTERNS: ClassVar[frozenset[str]] = frozenset(
-        [
-            r"|".join(
-                rf"^/{path}(\?.*)?$"
-                for path in (
-                    "(assets|events|files|timeseries|sequences|datasets|relationships|labels)/(list|byids|search|aggregate)",
-                    "files/downloadlink",
-                    "timeseries/(data(/(list|latest|delete))?|synthetic/query)",
-                    "sequences/data(/(list|delete))?",
-                    "raw/dbs/[^/]+/tables/[^/]+/rows(/delete)?",
-                    "context/entitymatching/(byids|list|jobs)",
-                    "sessions/revoke",
-                    "models/.*",
-                    ".*/graphql",
-                    "units/.*",
-                    "annotations/(list|byids|reverselookup)",
-                    r"functions/(list|byids|status|schedules/(list|byids)|\d+/calls/(list|byids))",
-                    r"3d/models/\d+/revisions/\d+/(mappings/list|nodes/(list|byids))",
-                    "documents/(aggregate|list|search|content|status|passages/search)",
-                    "profiles/(byids|search)",
-                    "geospatial/(compute|crs/byids|featuretypes/(byids|list))",
-                    "geospatial/featuretypes/[A-Za-z][A-Za-z0-9_]{0,31}/features/(aggregate|list|byids|search|search-streaming|[A-Za-z][A-Za-z0-9_]{0,255}/rasters/[A-Za-z][A-Za-z0-9_]{0,31})",
-                    "transformations/(filter|byids|jobs/byids|schedules/byids|query/run)",
-                    "simulators/list",
-                    "extpipes/(list|byids|runs/list)",
-                    "workflows/.*",
-                    "hostedextractors/.*",
-                    "postgresgateway/.*",
-                    "context/diagram/.*",
-                    "ai/tools/documents/(summarize|ask)",
-                    "ai/agents(/(byids|delete))?",
-                )
-            )
-        ]
+    __NON_RETRYABLE_CREATE_DELETE_RESOURCE_PATHS: ClassVar[list[str]] = [
```

Comment:
It looks like some paths that are not explicitly in the allow list (`datamodels` and others) are also not explicitly in the deny list. Is this intentional?

Reply:
Added a test to make sure we're matching the right APIs.

Reply:
But datamodels is in the allow list and intentionally not in the deny list.